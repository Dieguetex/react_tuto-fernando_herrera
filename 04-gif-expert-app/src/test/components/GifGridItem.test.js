import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas de <GifGridItem />', () => {
	const title = 'Un título';
	const url = 'https://localhost/algo.jgp';
	let wrapper = shallow(<GifGridItem title={title} url={url} />);

	test('Debe mostrar < GifGridItem /> correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe de tener un párrafo con el title', () => {
		const p = wrapper.find('p');
		expect(p.text().trim()).toBe(title);
	});

	test('Debe de tener la imágen igual al url y alt de los props', () => {
		const img = wrapper.find('img');

		expect(img.prop('src')).toBe(url);
		expect(img.prop('alt')).toBe(title);
	});

	test('Debe tener animate__fadeIn', () => {
		const div = wrapper.find('div');

		expect(div.hasClass('animate__fadeIn')).toBe(true);
	});
});
