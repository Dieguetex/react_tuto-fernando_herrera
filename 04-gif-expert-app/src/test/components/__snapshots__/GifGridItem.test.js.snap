// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem /> Debe mostrar < GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jgp"
  />
  <p>
    Un título
  </p>
</div>
`;
